packages/table/test/table.test.ts

```diff
@@ -0,0 +1,1043 @@
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+        expect(el.selected.length).to.equal(1);
+
+        await oneEvent(el, 'rangeChanged');
+        await elementUpdated(el);
+
+        const rowOneItem = el.querySelector('[value="0"]') as TableRow;
+        expect(rowOneItem.value).to.equal('0');
+    });
+
+    it('selects user-passed values for .selected array with no [selects="multiple"] specified, but does not allow interaction afterwards', async () => {
+        const el = await fixture<Table>(html`
+            <sp-table size="m" .selected=${['row2', 'row3']}>
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+                <sp-table-body style="height: 120px">
+                    <sp-table-row value="row1" class="row1">
+                        <sp-table-cell>Row Item Alpha</sp-table-cell>
+                        <sp-table-cell>Row Item Alpha</sp-table-cell>
+                        <sp-table-cell>Row Item Alpha</sp-table-cell>
+                    </sp-table-row>
+                    <sp-table-row value="row2" class="row2">
+                        <sp-table-cell>Row Item Bravo</sp-table-cell>
+                        <sp-table-cell>Row Item Bravo</sp-table-cell>
+                        <sp-table-cell>Row Item Bravo</sp-table-cell>
+                    </sp-table-row>
+                    <sp-table-row value="row3" class="row3">
+                        <sp-table-cell>Row Item Charlie</sp-table-cell>
+                        <sp-table-cell>Row Item Charlie</sp-table-cell>
+                        <sp-table-cell>Row Item Charlie</sp-table-cell>
+                    </sp-table-row>
+                </sp-table-body>
+            </sp-table>
+        `);
+        await elementUpdated(el);
+
+        expect(el.selected.length).to.equal(2);
+
+        const rowThree = el.querySelector('.row3') as TableRow;
+        const rowTwo = el.querySelector('.row2') as TableRow;
+        const rowOne = el.querySelector('.row1') as TableRow;
+        const rowOneCheckbox = rowOne.querySelector('sp-table-checkbox-cell');
+
+        expect(rowThree.selected, 'third row selected').to.be.true;
+        expect(rowTwo.selected, 'second row selected').to.be.true;
+        expect(rowOne.selected, 'first row selected').to.be.false;
+        expect(rowOneCheckbox).to.be.null;
+    });
+
+    it('selects user-passed values for .selected array with no [selects="multiple"] specified on Virtualized `<sp-table>`, but does not allow interaction afterwards', async () => {
```

Comment:
We likely want an edge case check that includes a consumer developer placing a checkbox, or other `change` event dispatching element in their own content and ensure in various contexts that it is handled correctly.